platform/src/test/java/org/exoplatform/platform/qa/ui/platform/answer/AnswerBasicTestsWithUserTestIT.java

```diff
@@ -0,0 +1,406 @@
+
+import org.exoplatform.platform.qa.ui.answer.pageobject.AnswerCategoryManagement;
+import org.exoplatform.platform.qa.ui.answer.pageobject.AnswerManagement;
+import org.exoplatform.platform.qa.ui.answer.pageobject.CommentManagement;
+import org.exoplatform.platform.qa.ui.answer.pageobject.QuestionManagement;
+import org.exoplatform.platform.qa.ui.commons.Base;
+import org.exoplatform.platform.qa.ui.core.PLFData;
+import org.exoplatform.platform.qa.ui.selenium.platform.HomePagePlatform;
+import org.exoplatform.platform.qa.ui.selenium.platform.ManageLogInOut;
+
+/**
+ * Created by ilyes on 07/09/17.
+ */
+public class AnswerBasicTestsWithUserTestIT extends Base {
+  HomePagePlatform         homePagePlatform;
+
+  AnswerManagement         answerManagement;
+
+  CommentManagement        commentManagement;
+
+  QuestionManagement       questionManagement;
+
+  AnswerCategoryManagement answerCategoryManagement;
+
+  ManageLogInOut           manageLogInOut;
+
+  @BeforeEach
+  public void setupBeforeMethod() {
+    info("Start setUpBeforeMethod");
+    homePagePlatform = new HomePagePlatform(this);
+    answerManagement = new AnswerManagement(this);
+    commentManagement = new CommentManagement(this);
+    questionManagement = new QuestionManagement(this);
+    answerCategoryManagement = new AnswerCategoryManagement(this);
+    manageLogInOut = new ManageLogInOut(this);
+    if ($(ELEMENT_SKIP_BUTTON).is(Condition.exist)) {
+      $(ELEMENT_SKIP_BUTTON).click();
+    }
+    if ($(ELEMENT_INPUT_USERNAME_CAS).is(Condition.not(Condition.exist))) {
+      manageLogInOut.signOut();
+    }
+    manageLogInOut.signInCas(PLFData.DATA_USER1, "gtngtn");
+  }
+
+  @AfterEach
+  public void signout() {
+    manageLogInOut.signOut();
+  }
+
+  /**
+   * Case ID:116817. Test Case Name: Answer a question. Case ID:116818. Test Case
+   * Name: Edit an answer. Case ID:116819. Test Case Name: Delete an answer.
+   * Pre-Condition: Post-Condition:
+   */
+  @Test
+  public void test04_AddAnswerAQuestion() {
+    String answer = "answer" + getRandomNumber();
+    String newanswer = "newanswer" + getRandomNumber();
+    String question = "question" + getRandomNumber();
+    String fullName = "fullName" + getRandomNumber();
+    String content = "content" + getRandomNumber();
+
+    info("Test 4: Answer a question");
+    /*
+     * Step Number: 1 Step Name: Prepare data: Create a category, a question, Step
+     * Description: - Login as john - Go to Answers - Create a category - Create a
+     * question in this category Input Data: Expected Outcome: - Category is created
+     * successfully. - Question is created successfully
+     */
+
+    /*
+     * Step number: 2 Step Name: Answer a question Step Description: - Open that
+     * question -Click on Answer, - Put data into editor - Set other fields - Save
+     * Input Data: Expected Outcome: - Answer a question successfully, - Show the
+     * answer below this question
+     */
```

Comment:
Move javadoc on the method and merge them.

Reply:
ok (y)